include/lsst/afw/typehandling/python.h

```diff
@@ -51,7 +51,12 @@ class StorableHelper : public Base {
 public:
     using Base::Base;
 
-    // Can't wrap clone(); PYBIND11_OVERLOAD chokes on unique_ptr return type
+    std::shared_ptr<Storable> cloneStorable() const override {
+        /* __deepcopy__ takes an optional dict, but PYBIND11_OVERLOAD_* doesn't
+         * allow translation between different argument lists in C++ and Python
+         */
```

Comment:
Are you saying this will fail if `__deepcopy__` is called with that argument?  (Not unreasonable - using it would at least be significantly more complex, if it's even possible - just asking for clarification in the comment.)

Reply:
I'm not sure if this answers your question, but the macro will not compile if you give it arguments (like `py::none()` or `py::dict()`) that are not present in `cloneStorable`'s argument list.